### ISSUE TITLE
Change AttributeAdapter classes access modifiers as public

### ISSUE THREAD

**Issue by khosro** [TanvirArjel/CustomValidation#9]
Thanks @ExLuzZziVo  for this great project.
I think it would be better to declare all AttributeAdapter classes as public, then we can use them in our custom impl of ```ValidationAttributeAdapterProvider```.
Because maybe we have other custom attributes and also we want to use your custom attributes, and for localization we must have our ```ValidationAttributeAdapterProvider```, then we can not use your impl of  ```ValidationAttributeAdapterProvider```.

**Comment by TanvirArjel** [comment]
@khosro Thank you so much for your suggestion. Let me think more about this.

**Comment by khosro** [comment]
@TanvirArjel Suppose that for file type Rar, browser sends ```application/octet-stream``` as you declare in https://github.com/TanvirArjel/CustomValidation/blob/master/src/TanvirArjel.CustomValidation.AspNetCore/Attributes/FileAttribute.cs#L120 , then in line https://github.com/TanvirArjel/CustomValidation/blob/master/src/TanvirArjel.CustomValidation.AspNetCore/Attributes/FileAttribute.cs#L233 , ```validFileTypes``` has values something like as follows :

```
...
[" application/octet-stream"]
...
```
pay attention to space in array.
then in line https://github.com/TanvirArjel/CustomValidation/blob/master/src/TanvirArjel.CustomValidation.AspNetCore/Attributes/FileAttribute.cs#L235 , if condition return false, but it must be return true.

The other way is using ```Trim``` when assign value to ```validFileTypes ``` as follows:

```
 string[] validFileTypes = FileTypes.Select(ft => ft.ToDescriptionString().ToUpperInvariant().Trim()).ToArray();

```

**Comment by TanvirArjel** [comment]
@khosro Thank you so much for noticing this! I shall add a fix to the next version. I could not merge your PR because that project has been deleted!

**Comment by TanvirArjel** [comment]
@khosro Please open a separate issue for this.

**Comment by khosro** [comment]
> @khosro Please open a separate issue for this.

**Comment by khosro** [comment]
>  @khosro Please open a separate issue for this.

Excuse me, i made a wrong answer here. I must answer in this [PR#8](https://github.com/TanvirArjel/CustomValidation/pull/8) .

**Comment by sgdowney** [comment]
Regarding the original issue, an alternative would be to make your implementation of the `IValidationAttributeAdapterProvider` public and leave everything else unchanged.  In my asp.net core application I want to use multiple 3rd party validation libraries.  I do this be creating and registering my own implementation of the `IValidationAttributeAdapterProvider` that then calls the `IValidationAttributeAdapterProvider` implementations of the 3rd party validation libraries.  Source below:
```
 public class CustomValidationAttributeAdapterProvider : IValidationAttributeAdapterProvider
    {
        public IAttributeAdapter GetAttributeAdapter(ValidationAttribute attribute, IStringLocalizer stringLocalizer)
        {
            IValidationAttributeAdapterProvider gocProvider = new GoC.Components.DataAnnotations.Provider.CustomValidationAttributeAdapterProvider();

            IAttributeAdapter adapter = gocProvider.GetAttributeAdapter(attribute, stringLocalizer);
            if (adapter == null)
            {
                IValidationAttributeAdapterProvider tanvirProvider = new TanvirArjel.CustomValidation.AspNetCore.AdapterProviders.CutomValidationAttributeAdapterProvider();
                adapter = tanvirProvider.GetAttributeAdapter(attribute, stringLocalizer);
            }
            return adapter;
        }
    }
```

**Comment by TanvirArjel** [comment]
@sgdowney Thank you for your suggestion, It has been implemented and will be available in version **1.2.0**;

**Comment by TanvirArjel** [comment]
Write your own **AttributeAdapterProvider** as follows:

```C#
public class YourCustomValidationAttributeAdapterProvider : TanvirArjelAttributeAdapterProvider
{
    public override IAttributeAdapter GetAttributeAdapter(ValidationAttribute attribute, IStringLocalizer stringLocalizer)
    {
        switch (attribute)
        {
            case FooAtrribute fooAtrribute:
                return new FooAtrributeAdapter(fooAtrribute, stringLocalizer);
            case CompareToAttribute compareToAttribute:
                return new MyCompareToAttributeAdapter(compareToAttribute, stringLocalizer);
            default:
                return base.GetAttributeAdapter(attribute, stringLocalizer);
        }
    }
}
```
Then register as follows:

```C#
services.AddSingleton<IValidationAttributeAdapterProvider, YourCustomValidationAttributeAdapterProvider>();
```
